### PR TITLE
Feature/Integrate Great Expectations into Airflow DAGs

### PR DESCRIPTION
**Pull request by AnthonyByansi**
**_WHAT DOES THIS PR DO?_**
This PR integrates Great Expectations (GE) into our Apache Airflow DAGs to automate data quality checks.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - OPS-226 OPS-227

## Summary by CodeRabbit

- **New Features**
  - Introduced extensive data validation steps for air quality measurements, including schema validation, data quality, uniqueness integrity, temporal consistency, completeness, referential integrity, and range validation.
  
- **Documentation**
  - Updated prerequisites and setup instructions for environment variables in the README.

- **Chores**
  - Enabled XCom pickling in the Dockerfile to facilitate data sharing between tasks in Apache Airflow.

### CODE REVIEW

File: src/workflows/dags/airqo_measurements.py [hunk 1]
```diff
@@ -171,13 +249,57 @@
...
+    validate_raw_data_quality = GreatExpectationsOperator(
+        task_id="validate_raw_air_quality_data_quality",
+        expectation_suite_name="raw_air_quality_data_quality",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "AirQo-dataset",
+            "table": "temp_raw_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:280]: It's been resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/9075cfe0f9978a6d564e8eb4dea279484329d7f4

File: src/workflows/dags/airqo_measurements.py [hunk 2]
```diff
@@ -171,13 +249,57 @@
...
+    validate_raw_uniqueness_integrity = GreatExpectationsOperator(
+        task_id="validate_raw_air_quality_uniqueness_integrity",
+        expectation_suite_name="raw_air_quality_uniqueness_integrity",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "AirQo-dataset",
+            "table": "temp_raw_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:291]: It's been resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/9075cfe0f9978a6d564e8eb4dea279484329d7f4

File: src/workflows/dags/airqo_measurements.py [hunk 3]
```diff
@@ -171,13 +249,57 @@
...
+
+    validate_raw_temporal_consistency = GreatExpectationsOperator(
+        task_id="validate_raw_air_quality_temporal_consistency",
+        expectation_suite_name="raw_air_quality_temporal_consistency",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "AirQo-dataset",
+            "table": "temp_raw_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:301]: It's been resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/9075cfe0f9978a6d564e8eb4dea279484329d7f4

File: src/workflows/dags/airqo_measurements.py [hunk 4]
```diff
@@ -171,13 +249,57 @@
...
+    validate_raw_schema = GreatExpectationsOperator(
+        task_id="validate_raw_air_quality_schema",
+        expectation_suite_name="raw_air_quality_schema",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "AirQo-dataset",
+            "table": "temp_raw_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:270]: It's been resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/9075cfe0f9978a6d564e8eb4dea279484329d7f4

File: src/workflows/dags/airqo_measurements.py [hunk 5]
```diff
@@ -171,13 +249,114 @@
...
+    validate_raw_temporal_consistency = GreatExpectationsOperator(
+        task_id='validate_raw_air_quality_temporal_consistency',
+        expectation_suite_name='raw_air_quality_temporal_consistency',
+        batch_kwargs={
+            'datasource': 'bigquery_datasource',
+            'dataset': 'AirQodataset',
+            'table': 'temp_raw_air_quality_data'
+        },
+        data_context_root_dir='gx/expectations'
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 6]
```diff
@@ -107,11 +109,87 @@
...
+    validate_temporal_consistency = GreatExpectationsOperator(
+        task_id="validate_air_quality_temporal_consistency",
+        expectation_suite_name="air_quality_temporal_consistency",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "AirQo_dataset",
+            "table": "temp_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:155]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 7]
```diff
@@ -171,13 +249,114 @@
...
+    
+    validate_raw_uniqueness_integrity = GreatExpectationsOperator(
+        task_id='validate_raw_air_quality_uniqueness_integrity',
+        expectation_suite_name='raw_air_quality_uniqueness_integrity',
+        batch_kwargs={
+            'datasource': 'bigquery_datasource',
+            'dataset': 'AirQodataset',
+            'table': 'temp_raw_air_quality_data'
+        },
+        data_context_root_dir='gx/expectations'
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 8]
```diff
@@ -107,11 +109,87 @@
...
+    validate_referential_integrity = GreatExpectationsOperator(
+        task_id="validate_air_quality_referential_integrity",
+        expectation_suite_name="air_quality_referential_integrity",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "AirQodataset",
+            "table": "temp_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:177]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 9]
```diff
@@ -171,13 +249,114 @@
...
+    validate_raw_referential_integrity = GreatExpectationsOperator(
+        task_id='validate_raw_air_quality_referential_integrity',
+        expectation_suite_name='raw_air_quality_referential_integrity',
+        batch_kwargs={
+            'datasource': 'bigquery_datasource',
+            'dataset': 'AirQodataset',
+            'table': 'temp_raw_air_quality_data'
+        },
+        data_context_root_dir='gx/expectations'
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 10]
```diff
@@ -171,13 +249,114 @@
...
+    validate_raw_range = GreatExpectationsOperator(
+        task_id='validate_raw_air_quality_range_validation',
+        expectation_suite_name='raw_air_quality_range_validation',
+        batch_kwargs={
+            'datasource': 'bigquery_datasource',
+            'dataset': 'AirQodataset',
+            'table': 'temp_raw_air_quality_data'
+        },
+        data_context_root_dir='gx/expectations'
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 11]
```diff
@@ -171,13 +249,114 @@
...
+    validate_raw_completeness = GreatExpectationsOperator(
+        task_id='validate_raw_air_quality_completeness',
+        expectation_suite_name='raw_air_quality_completeness',
+        batch_kwargs={
+            'datasource': 'bigquery_datasource',
+            'dataset': 'AirQodataset',
+            'table': 'temp_raw_air_quality_data'
+        },
+        data_context_root_dir='gx/expectations'
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

File: src/workflows/dags/airqo_measurements.py [hunk 12]
```diff
@@ -107,11 +109,87 @@
...
+    validate_schema = GreatExpectationsOperator(
+        task_id="validate_air_quality_schema",
+        expectation_suite_name="air_quality_schema_validation",
+        batch_kwargs={
+            "datasource": "bigquery_datasource",
+            "dataset": "Aiqo-dataset",
+            "table": "temp_air_quality_data",
+        },
+        data_context_root_dir="gx/expectations",
+    )
```
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:122]: Resolved through https://github.com/airqo-platform/AirQo-api/pull/3262/commits/f0c5f7d68a93b9e640a433a73a7d4eabff24dee8

**Review by NicholasTurner23 — APPROVED** [review]
Looks good.

File: src/workflows/dags/airqo_measurements.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/workflows/dags/airqo_measurements.py:280]: cc: @AnthonyByansi .......please review this feedback

**Review by Baalmart — COMMENTED** [review]
Thanks @AnthonyByansi , please review a pending review comment by CodeRabbit. Thanks.

**Review by MartinKalema — APPROVED** [review]

File: src/workflows/dags/airqo_measurements.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:280]: @coderabbitai
The variable `validate_raw_data_quality` has been used used here. Double check and let me know if I'm wrong!

```python
(
        extract_raw_data()
        >> [
            validate_raw_schema,
            validate_raw_data_quality,
            validate_raw_uniqueness_integrity,
            validate_raw_temporal_consistency,
            validate_raw_completeness,
            validate_raw_referential_integrity,
            validate_raw_range,
        ]
        >> clean_data_raw_data()
    )

```

[the remark this answers is not in the document]
Comment (AnthonyByansi) [reply src/workflows/dags/airqo_measurements.py:280]: @Baalmart It looks clean. Please have a look at the above comment by @coderabbitai

**Review by AnthonyByansi — COMMENTED** [review]
LGTM 🥂

**Review by Baalmart — APPROVED** [review]
thanks @AnthonyByansi

**Pull request merged**